e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,529 @@
+	require.ErrorContains(t, err, "minter address ≠ minter controller's minter address")
+
+	showMintersRes, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minters")
+	expectedShowMinters := fiattokenfactorytypes.QueryGetMintersResponse{
+		Minters: fiattokenfactorytypes.Minters{
+			Address: minter1.FormattedAddress(),
+			Allowance: sdktypes.Coin{
+				Denom:  "uusdc",
+				Amount: math.NewInt(allowance),
+			},
+		},
+	}
+	require.Equal(t, expectedShowMinters.Minters, showMintersRes.Minters, "configured minter allowance is not as expected")
+
+	// ACTION: Configure a minter that is blacklisted
+	// EXPECTED: Success; Minter is configured with allowance
+	// Status:
+	// minterController1 -> minter1
+	// gw minterController -> gw minter
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, minter1)
+
+	configureMinter(t, ctx, val, minterController1, minter1, 11)
+}
+
+func TestFiatTFRemoveMinter(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	t.Parallel()
+
+	ctx := context.Background()
+
+	nw := nobleSpinUp(t, ctx, true)
+	noble := nw.chain
+	val := noble.Validators[0]
+
+	// ACTION: Remove minter while TF is paused
+	// EXPECTED: Success; Minter is removed
+
+	allowance := int64(10)
+
+	pauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.MinterController.KeyName(), "fiat-tokenfactory", "remove-minter", nw.fiatTfRoles.Minter.FormattedAddress())
+	require.NoError(t, err, "error broadcasting removing minter")
+
+	_, err = showMinters(ctx, val, nw.fiatTfRoles.Minter)
+	require.Error(t, err, "minter found; not successfully removed")
+
+	unpauseFiatTF(t, ctx, val, nw.fiatTfRoles.Pauser)
+
+	// ACTION: Remove minter from a minter controller not associated with the minter
+	// EXPECTED: Request fails; Minter is not removed
+	// Status:
+	// 	gw minterController -> gw minter (Removed)
+
+	// reconfigure minter
+	configureMinter(t, ctx, val, nw.fiatTfRoles.MinterController, nw.fiatTfRoles.Minter, allowance)
+
+	minter1, _ := setupMinterAndController(t, ctx, noble, val, nw.fiatTfRoles.MasterMinter, allowance)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MinterController.KeyName(), "fiat-tokenfactory", "remove-minter", minter1.FormattedAddress())
+	require.ErrorContains(t, err, "minter address ≠ minter controller's minter address")
+
+	// ensure minter still exists
+	showMintersRes, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minter")
+	expectedShowMinters := fiattokenfactorytypes.QueryGetMintersResponse{
+		Minters: fiattokenfactorytypes.Minters{
+			Address: minter1.FormattedAddress(),
+			Allowance: sdktypes.Coin{
+				Denom:  "uusdc",
+				Amount: math.NewInt(allowance),
+			},
+		},
+	}
+	require.Equal(t, expectedShowMinters.Minters, showMintersRes.Minters)
+
+	// ACTION: Remove minter from a blacklisted minter controller
```

Comment:
for completeness, can we blacklist both minter and minter controller in this test?

Reply:
Added in: https://github.com/noble-assets/noble/pull/368/commits/ec68a678ca5965a3132107b1879e481ff04972af